staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -437,3 +437,43 @@ func resetOrReuseTimer(t *time.Timer, d time.Duration, sawTimeout bool) *time.Ti
 	t.Reset(d)
 	return t
 }
+
+// WaitFuncUntil waits a given duration for 'fn' to return.
+//
+// WaitFuncUntil creates two channels to determine whether if
+// it reaches to timeout or 'fn' returned
+//
+// If 'fn' returns an error the error is returned, and if 'fn'
+// returns ture nil is returned
+//
+// It's caller's duty to make sure 'fn' will return at last.
+func WaitFuncUntil(d time.Duration, fn ConditionFunc) error {
```

Comment:
I think in golang it's easy enough to timeout a function, for example, https://github.com/golang/go/wiki/Timeouts. I suggest just rewriting the `PathExists` function with a channel.

Reply:
How about write a new function wrapping `PathExists` like `PathExistsWithTimeout`, since PathExists was a straightforward function and already used everywhere

Reply:
Yeah, that makes sense.

Reply:
I meant to say `WaitFuncUntil` isn't worth a library function. I suggest moving all the logic to `PathExistsWithTimeout`.

Reply:
Will do. 

It's indeed a small function, I was thinking wrapping it as a separated function would make it easier to write test cases while it's difficult to mock a timeout result for `PathExists`.